frontend/components/Editor.js

```diff
@@ -463,6 +463,24 @@ export class Editor extends Component {
                 const delta = before_or_after == "before" ? 0 : 1
                 return await this.actions.add_remote_cell_at(index + delta, code)
             },
+            // These actions are basically helpers because pushing selected cells all
+            // the way down is too heavy to handle!
+            fold_code: async (cell_id, selected) => {
```

Comment:
All these new functions are optimizations. Instead of doing the math in each cell, we use this functions instead. That means we can only have one function (this one) that needs to refer to `this`.`state`.`selected_cells` and we don't need to re-memo each time! (we did that once per cell🙈)